homeassistant/components/mqtt/siren.py

```diff
@@ -379,6 +387,8 @@ def _update(self, data: SirenTurnOnServiceParameters) -> None:
         """Update the extra siren state attributes."""
         for attribute, support in SUPPORTED_ATTRIBUTES.items():
             if self._attr_supported_features & support and attribute in data:
-                self._extra_attributes[attribute] = data[
-                    attribute  # type: ignore[literal-required]
-                ]
+                data_attr = data[attribute]  # type: ignore[literal-required]
+                if self._extra_attributes.get(attribute) == data_attr:
+                    continue
+                self._dirty += 1
+                self._extra_attributes[attribute] = data_attr
```

Comment:
Can this be moved to a separate PR?

Reply:
https://github.com/home-assistant/core/pull/100871